neo/Network/P2P/TaskManager.cs

```diff
@@ -272,5 +277,13 @@ internal protected override bool IsHighPriority(object message)
                     return false;
             }
         }
+
+        internal protected override bool ShallDrop(object message, IEnumerable queue)
+        {
+            if (!(message is TaskManager.NewTasks tasks)) return false;
+            // Remove duplicate tasks
+            if (queue.OfType<TaskManager.NewTasks>().Any(x => x.Payload.Type == tasks.Payload.Type && x.Payload.Hashes.SequenceEqual(tasks.Payload.Hashes))) return true;
```

Comment:
I am worried that this will lead to blockage.

Can we just limit the length of the mailbox, and if it exceeds a certain length, we will abandon the new package directly?

Reply:
What do you think about create a merkle tree for make this faster?

Reply:
In NEO3, we can make each `InvPayload` only contain one hash, because in fact we have almost no usage of multiple hashes in one `InvPayload`.

In NEO2, we urgently need to solve the empty block and the sync stuck problem, so I think that no matter what method can be used as long as it can solve the problem.

Reply:
My proposal is use this patch, but with a merkle tree

Reply:
> In NEO3, we can make each InvPayload only contain one hash, because in fact we have almost no usage of multiple hashes in one InvPayload.

There are some lines that are used like 

https://github.com/neo-project/neo/blob/cd994894acce73a871dccdf2ba6c2f0d598ec1b8/neo/Consensus/ConsensusService.cs#L445

Reply:
Shargon, currently this will cause also an attack, even with this fix PR, right?